### PR TITLE
Make tile element direction accessible for scripts

### PR DESCRIPTION
**Pull request by tmstieff**
Add the direction field from the tile element class to the ScTileElement class passed to the

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
You can't set a large tile element or park entrance rotation. It will really mess up a bunch of expectations. It's the reason you can't set them from the tile inspector.

**Comment by Gymnasiast** [comment]
We should perhaps create a function `CanRotate()` on `TileElementBase` and have both the Tile Inspector and this new function call it.

**Comment by IntelOrca** [comment]
I would only define `direction` on the tile element types that support it within `openrct2.d.ts`. Then as @duncanspumpkin said, make sure it only read / writes direction for those valid element types.

**Comment by duncanspumpkin** [comment]
Reading the direction is still useful for all types. It's just the setting which is an issue.

**Comment by tmstieff** [comment]
I think I would have to defer this feature to someone with more knowledge of the game data structures.

**Comment by duncanspumpkin** [comment]
I did some more investigating and tracks seem to handle this safely after all. You can get in some weird situations with tracks but the game just prevents you from doing anything. For large scenery it will definitely cause issues if you rotate and delete parts so I would want large scenery to not have rotation. Park entrances I doubt anyone will notice too much rotation related issues.

File: distribution/openrct2.d.ts [hunk 1]
```diff
@@ -449,6 +449,8 @@ declare global {
 
         addition: number | null;
         isAdditionBroken: boolean;
+
+        direction: 0 | 1 | 2 | 3;
```
Comment (IntelOrca) [inline distribution/openrct2.d.ts:453]: It might be nice to declare a type for this now. E.g.:
`type Direction = 0 | 1 | 2 | 3`

Comment (tmstieff) [reply distribution/openrct2.d.ts]: Good call, definitely clearer.

**Comment by duncanspumpkin** [comment]
Is there no way to have the property readable but not writeable for the problem entries? I know at least one plugin developer was wanting to read the rotations for all elements.

**Comment by tmstieff** [comment]
I can add a readonly direction field to the other elements. Do you know if it works as expected for the large scenery and path elements?

**Comment by IntelOrca** [comment]
@duncanspumpkin yes, you can pass nullptr to the setter method.

File: src/openrct2/scripting/ScTile.hpp [hunk 2]
```diff
@@ -1013,6 +1032,7 @@ namespace OpenRCT2::Scripting
                 ctx, &ScTileElement::isAdditionBroken_get, &ScTileElement::isAdditionBroken_set, "isAdditionBroken");
             dukglue_register_property(
                 ctx, &ScTileElement::isAdditionGhost_get, &ScTileElement::isAdditionGhost_set, "isAdditionGhost");
+            dukglue_register_property(ctx, &ScTileElement::direction_get, &ScTileElement::direction_set, "direction");
```
Comment (Broxzier) [inline src/openrct2/scripting/ScTile.hpp:1035]: This is added under the "Footpath only" section, but applies to all element types, right? This should be moved up to be listed under "All".

Comment (tmstieff) [reply src/openrct2/scripting/ScTile.hpp:1035]: I missed that, I'll move it up.

**Pull request merged**